index.js

```diff
@@ -1,11 +1,28 @@
 var N = require('libnested')
 
 var isModule = require('./is')
+var apply = require('./apply')
+
+module.exports = function combine () {
+  var modules = flattenNested(arguments)
+
+  throwIfMissingDeps(modules)
 
-function hasAll(set, keys) {
-  return keys.every(function (k) {
-    return !!set[k]
+  var api = {}
+
+  eachModule(modules, function (module, key) {
+    var deps = buildDeps(module.needs, api)
+    var exported = module.create(deps)
+
+    if (!exported) { throw new Error('export declared but not returned for', key) }
```

Comment:
also, how are we distinguishing between key and path? 
I think this could be `path`?